esprima.js

```diff
@@ -1948,11 +1948,12 @@
             return this;
         },
 
-        finishTryStatement: function (block, guardedHandlers, handlers, finalizer) {
+        finishTryStatement: function (block, handler, finalizer) {
             this.type = Syntax.TryStatement;
             this.block = block;
-            this.guardedHandlers = guardedHandlers;
-            this.handlers = handlers;
+            this.guardedHandlers = [];
+            this.handlers = handler ? [ handler ] : [];
```

Comment:
Yes, but perhaps we need to introduce backCompat mode for this. Because printing an obsolete warning is unexpected output. Unless you are saying print obsolete notice in 3.0.0 and remove in 4.0.0, but that seems a bit too cautious.